vortex-layout/src/macros.rs

```diff
@@ -0,0 +1,24 @@
+/// A macro for optionally instrumenting a future, if tracing feature is enabled.
+#[macro_export]
```

Comment:
I can see `vortex-tracing` forming 😂 
personally I like the postfix pattern better because it composes better, but nbd.

Reply:
Yea I like postfix more as well, but how would you pass span args then because you would have a rust function not a macro as your entrypoint. We could accept the span! as an argument like tracing but then you need to optionally use it at calsite